python/src/aiconfig/router.py

```diff
@@ -0,0 +1,24 @@
+from abc import ABCMeta, abstractmethod
+from typing import List
+from aiconfig import Prompt
+from aiconfig.AIConfigSettings import Output
+from aiconfig.Config import AIConfigRuntime
+
+class Router(Metaclass=ABCMeta):
+
+    def __init__(self, config: AIConfigRuntime):
+        self.config = config
+
+    @abstractmethod
+    def find(input, **kwargs) -> List[Prompt]:
+        """
+        Search and select a list of suitable prompts
+        """
+        pass
+
+    @abstractmethod
+    def route(self, **kwargs) -> Output:
```

Comment:
This also needs the input to use to route with.
```suggestion
    def route(self, input, **kwargs) -> Output:
```